forum/qa-plugin/ckeditor4/qa-ckeditor4-layer.php

```diff
@@ -78,21 +78,25 @@ public function head_css()
         qa_html_theme_base::head_css();
     }
 
-    public function head_script()
+    public function main()

```

Comment:
Nazwa metody `main` jest moim zdaniem trochę bezkontekstowa (w porównaniu do poprzedniej nazwy `head_script`) - może nazwać ją `main_script`?

Reply:
Prosto nie można, bo to jest nadpisanie - to jest moduł layer, który rozszerza qa_html_theme_base. W tej klasie jest metoda main() i tam było dodane podpięcie init.js na sztywno. Ja to wyrzuciłem i nadpisałem tutaj metodę, najpierw wykonując to co jest w tym pierwotnym parencie (`parent::main();`) a dalej dodając swoje skrypty. Jeśli zmienimy nazwę metody na taką której nie ma w klasie bazowej/której Q2A nie wywołuje gdzieś tam sobie no to nie uruchomi się, trzeba by kombinować. Można tu oczywiscie stworzyć nową metodę którą tylko się wywoła w main albo inne podobne rzeczy. Widzę też, że jest metoda `body_suffix()` (https://github.com/CodersCommunity/forum.pasja-informatyki.local/blob/22dc993f6b2002fe4e67ae62137d989e55c3f4a4/forum/qa-include/qa-theme-base.php#L427), może można spróbować ją nadpisać i tam dodać te skrypty i będzie czytelniej

Reply:
przeniosłem, wygląda na to że działa i teraz zamiast za main dodaje prawie na samym końcu body

Reply:
Jeśli masz na myśli skrypt "/qa-plugin/ckeditor4/plugins/syntaxhighlight/init.js", to on celowo miał być podpinany tuż za `<main>`, aby dekorowanie postów (bloczki z kodem + snippety) było odpalane od razu po tym, jak posty pojawią się w DOM. ;) Z dwojga złego lepsza będzie bezkontekstową nazwa metody. ;)

Reply:
ok, w takim razie przywróciłem jak było. Pewnie by się dało to zrobić lepiej, ale nie wiem czy jest kombinować, skoro Q2A takie nadpisywanie ma od razu wbudowane.

Reply:
Fajnie. Pomyślałem, że można by zakombinować coś takiego
`public function main() {
  return function main_script() {
      // ...
  };
}`
, wtedy struktura by była zachowana i nazewnictwo bardziej zrozumiałe... ale chyba nie ma sensu tak kombinować. ;)

Reply:
Nie da się tak, możesz zrobić closure i wtedy nie ma nazwy, coś w rodzaju:
```php
public function main()
{
    return fn() => 'test';
}
```
z tym że wywołanie `main()` nie da `test`, bo trzeba jeszcze wywołać samo closure, a więc byłoby `$object->main()();`, a to wymagałoby zmiany wywoływania, co się dzieje gdzieś na poziomie Q2A i pewnie nie chcemy się tym zajmować.

Jak bardzo chcesz to po prostu zrobiłbym dodatkową metodę i ją wywołał, coś typu:
```php
public function main()
{
    $this->main_script();
}

public function main_script()
{
    // ...
}
```